flask/packages.py

```diff
@@ -0,0 +1,145 @@
+
+    user = db.users.find_one({"uuid": uuid})
+
+    if not user:
+        return jsonify({"status": "error", "message": "Unauthorized"}), 401
+
+    if request.method == "POST":
+        name = request.form.get("name")
+        namespace = request.form.get("namespace")
+        tarball  = request.files['tarball']
+        version = request.form.get("version")
+        license = request.form.get("license")
+        copyright = request.form.get("copyright")
+        description = request.form.get("description")
+        namespace_description = request.form.get("namespace_description")
+        tags = request.form.get("tags").strip().split(",")
+        dependencies = request.form.get("dependencies").strip().split(",")
+        # for dependency in list(set(dependencies)):
+        #     dependencies_id = []
+        #     resp = db.packages.find_one({"name": dependency})
+        #     if resp:
+        #         dependencies_id.append(resp["_id"])
+
+        package = db.packages.find_one({"name": name, "version": version})
+        if package is not None:
+            return jsonify({"status": "error", "message": "Package already exists"}), 400
+        
+        tarball_name = "{}-{}.tar.gz".format(name, version)
+        tarball.save(tarball_name)
+
+        package = {
+            "name": name,
+            "namespace": namespace,
+            "tarball": tarball_name,
+            "version": version,
+            "license": license,
+            "createdAt": datetime.utcnow(),
+            "author": user["_id"],
+            "maintainers": [user["_id"]],
+            "copyright": copyright,
+            "description": description,
+            "tags": list(set(tags)),
+            "dependencies": dependencies,
+        }
+        db.packages.insert_one(package)
+
+        namespace_doc = db.namespaces.find_one({"namespace": namespace})
+        if namespace_doc:
+            namespace_doc["packages"].append(package["_id"])
+            db.namespaces.update_one(
+                {"_id": namespace_doc["_id"]}, {"$set": namespace_doc}
+            )
+        else:
+            namespace_doc = {
+                "namespace": namespace,
+                "createdAt": datetime.utcnow(),
+                "createdBy": user["_id"],
+                "description": namespace_description,
+                "tags": tags,
+                "authors": user["_id"],
+                "packages": [package["_id"]],
+            }
+            db.namespaces.insert_one(namespace_doc)
+
+        # user["authorOf"].append(package["_id"])
+        db.users.update_one({"_id": user["_id"]}, {"$set": user})
+
+        return jsonify({"message": "Package Uploaded Successfully.", "code": 200})
+
+
+@app.route("/packages/<namespace_name>/<package_name>", methods=["GET"])
+def get_package(namespace_name, package_name):
+    # Get package from a package_name and namespace_name.
+    package = db.packages.find_one({"name": package_name, "namespace": namespace_name})
```

Comment:
we would have to search for the namespace 's `_id` then we would have to use it to search for a package.we should also consider to send only a specific set of details rather than the entire document from collection.